plumber-android/src/main/java/leakcanary/AndroidLeakFixes.kt

```diff
@@ -198,12 +205,223 @@ enum class AndroidLeakFixes {
+          activityChooserGetterMethod.isAccessible = true
+          val onChooseActivityListenerClass = activityChooserModelClass
+            .declaredClasses
+            .first { clazz -> clazz.simpleName == "OnChooseActivityListener" }
+          activityChooserSetterMethod = activityChooserModelClass
+            .getDeclaredMethod("setOnChooseActivityListener", onChooseActivityListenerClass)
+          activityChooserSetterMethod.isAccessible = true
+        } catch (ignored: Exception) {
+          return@execute
+        }
+
+        application.onActivityDestroyed {
+          try {
+            val activityChooser = activityChooserGetterMethod
+              .invoke(null, application, ShareActionProvider.DEFAULT_SHARE_HISTORY_FILE_NAME)
+            activityChooserSetterMethod.invoke(activityChooser, null)
+          } catch (ignored: Exception) {
+            // Silent
+          }
+        }
+      }
+    }
+  },
+
+  /**
+   * ConnectivityManager has a sInstance field that is set when the first ConnectivityManager instance is created.
+   * ConnectivityManager has a mContext field.
+   * When calling activity.getSystemService(Context.CONNECTIVITY_SERVICE) , the first ConnectivityManager instance
+   * is created with the activity context and stored in sInstance.
+   * That activity context then leaks forever.
+   *
+   * Tracked here: https://code.google.com/p/android/issues/detail?id=198852
+   * Introduced here: https://github.com/android/platform_frameworks_base/commit/e0bef71662d81caaaa0d7214fb0bef5d39996a69
+   */
+  CONNECTIVITY_MANAGER {
+    override fun apply(application: Application) {
+      if (SDK_INT > 23) {
+        return
+      }
+
+      try {
+        application.getSystemService(Context.CONNECTIVITY_SERVICE)
+      } catch (ignored: Exception) {
+        // Silent
+      }
+    }
+  },
+
+  /**
+   * ClipboardUIManager is a static singleton that leaks an activity context.
+   */
+  CLIPBOARD_MANAGER {
+    override fun apply(application: Application) {
+      if (MANUFACTURER != SAMSUNG || SDK_INT !in 19..21) {
+        return
+      }
+
+      try {
+        val managerClass = Class.forName("android.sec.clipboard.ClipboardUIManager")
+        val instanceMethod = managerClass.getDeclaredMethod("getInstance", Context::class.java)
+        instanceMethod.isAccessible = true
+        instanceMethod.invoke(null, application)
+      } catch (ignored: Exception) {
+        // Silent
+      }
+    }
+  },
+
+  /**
+   * The static method MediaScannerConnection.scanFile() takes an activity context
+   * but the service might not disconnect after the activity has been destroyed.
+   *
+   * Tracked here: https://code.google.com/p/android/issues/detail?id=173788
+   */
+  MEDIA_SCANNER {
+    override fun apply(application: Application) {
+      if (SDK_INT > 22) {
+        return
+      }
+
+      try {
+        MediaScannerConnection(application, null).apply {
+          connect()
```

Comment:
Is this fast? If not, should it be done from the background thread (and would that introduce additional risk?)

Reply:
`connect` and `disconnect` basically are just calls to `bindService` and `unbindService` so this should normally be pretty fast imho

Reply:
After re-reading the issue (that you've twitted a few days ago if I'm not mistaken), I've realized my fix does not address the issue, and the fix is to be done on the developper and not automatically, so I'm removing this part.